### PR TITLE
Added a debug message for wrong XML response in 'process_xml'

### PR DESCRIPTION
**Pull request by Kulgar**
Hi there :)

I'm making a small pull request to add some debug information when the gem is trying to process a bad xml response. 
Along with the error message I also add the original XML content so that we can know for sure what was the response and what is going on. 

I also added a spec to test this error message in outcome_response_spec.rb. 

I think it may benefit people to debug their apps. 

Let me know if I should modify my PR. 
Regis.

### CODE REVIEW

File: lib/ims/lti/outcome_response.rb [hunk 1]
```diff
@@ -105,18 +105,22 @@ def has_error?
...
+        @code_major.downcase! if @code_major
+        @severity = doc.text("//imsx_statusInfo/imsx_severity")
+        @severity.downcase! if @severity
+        @description = doc.text("//imsx_statusInfo/imsx_description")
+        @description = @description.to_s if @description
+        @message_ref_identifier = doc.text("//imsx_statusInfo/imsx_messageRefIdentifier")
+        @operation = doc.text("//imsx_statusInfo/imsx_operationRefIdentifier")
+        @score = doc.text("//readResultResponse//resultScore/textString")
+        @score = @score.to_s if @score
+      rescue => e
```
Comment (wbhumphrey) [inline lib/ims/lti/outcome_response.rb]: Can you scope the error to be specific to the XML parse error you are expecting?

**Comment by wbhumphrey** [comment]
What error were you getting previously?  Was it failing on doc = REXML::Document.new xml?

**Comment by Kulgar** [comment]
Hi, yep, here is the error I got:

```
#<RuntimeError: attempted adding second root element to document>
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/document.rb:95:in `add'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/element.rb:879:in `add'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/child.rb:21:in `initialize'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/parent.rb:13:in `initialize'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/element.rb:59:in `initialize'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/element.rb:877:in `new'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/element.rb:877:in `add'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/element.rb:297:in `add_element'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/document.rb:102:in `add_element'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/parsers/treeparser.rb:33:in `parse'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/document.rb:287:in `build'
/usr/local/rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/rexml/document.rb:44:in `initialize'
ruby/2.1.0/gems/ims-lti-1.1.7/lib/ims/lti/outcome_response.rb:108:in `new'
ruby/2.1.0/gems/ims-lti-1.1.7/lib/ims/lti/outcome_response.rb:108:in `process_xml'
ruby/2.1.0/gems/ims-lti-1.1.7/lib/ims/lti/outcome_response.rb:78:in `process_post_response'
ruby/2.1.0/gems/ims-lti-1.1.7/lib/ims/lti/outcome_request.rb:131:in `post_outcome_request'
ruby/2.1.0/gems/ims-lti-1.1.7/lib/ims/lti/outcome_request.rb:81:in `post_replace_result!'
ruby/2.1.0/gems/ims-lti-1.1.7/lib/ims/lti/tool_provider.rb:119:in `post_replace_result!'
```

**Comment by wbhumphrey** [comment]
I am okay with the idea here.  However, I do have a few things I would like you to do differently

* We should only rescue on line 109: `doc = REXML::Document.new xml`
* We should create a new error (XMLParseError?) and raise that instead of raising another StandardError
* Maybe shorten the message to just "#{e}\nOriginal XML:'#{xml}''"
* Change names with wrong to bad in the spec (ie wrong_xml -> bad_xml)
* The Travis CI build should pass

**Comment by Kulgar** [comment]
Ok! I'll take care of that :) Thanks!

**Comment by Kulgar** [comment]
Done :)

**Comment by rivernate** [comment]
Sorry it took so long

**Pull request merged**